compiler/kernel/bufferfilter.go

```diff
@@ -63,7 +66,11 @@ func CompileBufferFilter(e dag.Expr) (*expr.BufferFilter, error) {
 			return expr.NewOrBufferFilter(left, right), nil
 		}
 		left := expr.NewBufferFilterForStringCase(e.Text)
-		right, err := newBufferFilterForLiteral(e.Value)
+		literal, err := zson.ParseValue(zctx, e.Value)
+		if err != nil {
+			return nil, err
+		}
+		right, err := newBufferFilterForLiteral(&literal)
```

Comment:
I don't think there's any need to parse `e.Value` a second time.
```suggestion
		right, err := newBufferFilterForLiteral(&val)
```